apstra/api_device_profiles_modular.go

```diff
@@ -0,0 +1,127 @@
+package apstra
+
+import (
+	"context"
+	"fmt"
+	"github.com/orsinium-labs/enum"
+	"net/http"
+)
+
+type DeviceProfileType enum.Member[string]
+
+var (
+	DeviceProfileTypeModular    = DeviceProfileType{Value: "modular"}
+	DeviceProfileTypeMonolithic = DeviceProfileType{Value: "monolithic"}
+)
+
+type rawModularDeviceSlotConfiguration struct {
+	LinecardProfileId ObjectId `json:"linecard_profile_id"`
+	SlotId            uint64   `json:"slot_id"`
+}
+
+type ModularDeviceSlotConfiguration struct {
+	LinecardProfileId ObjectId
+}
+
+type ModularDeviceProfile struct {
+	Label              string
+	ChassisProfileId   ObjectId
+	SlotConfigurations map[uint64]ModularDeviceSlotConfiguration
+}
+
+func (o *ModularDeviceProfile) raw() *rawModularDeviceProfile {
+	result := &rawModularDeviceProfile{
+		DeviceProfileType:  DeviceProfileTypeModular.Value,
+		Label:              o.Label,
+		ChassisProfileId:   o.ChassisProfileId,
+		SlotConfigurations: make([]rawModularDeviceSlotConfiguration, len(o.SlotConfigurations)),
+	}
+
+	var i int
+	for slotId, slotConfiguraiton := range o.SlotConfigurations {
+		result.SlotConfigurations[i] = rawModularDeviceSlotConfiguration{
+			SlotId:            slotId,
+			LinecardProfileId: slotConfiguraiton.LinecardProfileId,
```

Comment:
spelling (configraiton)

Reply:
Spelling fixed in 67138882d2037fda5813b7eed54c0bb6597bdf01